crates/rustc_codegen_spirv/src/linker/inline.rs

```diff
@@ -350,6 +352,24 @@ impl Inliner<'_, '_> {
         result
     }
 
+    ///Applies all rewrite rules to the decorations in the header.
+    fn apply_rewrite_for_decorations(&mut self, rewrite_rules: &FxHashMap<Word, Word>) {
+        // NOTE(siebencorgie): We don't care *what* decoration we rewrite atm. AFAIK there is no case where rewriting
+        // the decoration on inline wouldn't be valid.
+        for annotation_idx in 0..self.annotations.len() {
+            if self.annotations[annotation_idx].class.opcode == Op::Decorate {
```

Comment:
Hmm this is enough for your usecase but it should be possible to extend it to all decorate instructions (which I'll need for late zombie reporting).